plugins/slicer/MONAILabel/MONAILabel.py

```diff
@@ -1258,7 +1282,13 @@ def onSaveLabel(self):
             label_in = tempfile.NamedTemporaryFile(suffix=self.file_ext, dir=self.tmpdir).name
             self.reportProgress(5)
 
-            slicer.util.saveNode(labelmapVolumeNode, label_in)
+            if (
+                slicer.util.settingsValue("MONAILabel/allowOverlappingSegments", True, converter=slicer.util.toBool)
+                and slicer.util.settingsValue("MONAILabel/fileExtension", self.file_ext) == ".seg.nrrd"
```

Comment:
`.seg.nrrd`? why we need to hard-fix this extension?

Reply:
1) .seg.nrrd makes sure all layers are preserved
2) .seg.nrrd makes sure that everything is exactly reproduced when label file is loaded back into Slicer

please see also discussion at: https://discourse.slicer.org/t/how-to-correctly-export-overlapping-segmentations/21050/

pynrrd and slicerio are useful python packages to read .seg.nrrd files